src/main/java/github/scarsz/discordsrv/DiscordSRV.java

```diff
@@ -712,43 +834,97 @@ private List<InetAddress> lookupPublic(String host) throws UnknownHostException
 
         Optional<Boolean> noopHostnameVerifier = config().getOptionalBoolean("NoopHostnameVerifier");
 
-        // Limit okhttp to 20 concurrent requests to avoid hogging every available thread
+        // Limit okhttp to 20 concurrent requests to avoid hogging every available
+        // thread
         Dispatcher dispatcher = new Dispatcher(
                 new ThreadPoolExecutor(
                         2, 20, 5, TimeUnit.SECONDS,
-                        new SynchronousQueue<>(), Util.threadFactory("OkHttp Dispatcher", false))
-        );
+                        new SynchronousQueue<>(), Util.threadFactory("OkHttp Dispatcher", false)));
         dispatcher.setMaxRequests(20);
         dispatcher.setMaxRequestsPerHost(20); // most requests are to discord.com
         ConnectionPool connectionPool = new ConnectionPool(5, 10, TimeUnit.SECONDS);
 
-        OkHttpClient httpClient = new OkHttpClient.Builder()
-                .dispatcher(dispatcher)
-                .connectionPool(connectionPool)
-                .dns(dns)
-                // more lenient timeouts (normally 10 seconds for these 3)
-                .connectTimeout(20, TimeUnit.SECONDS)
-                .readTimeout(20, TimeUnit.SECONDS)
-                .writeTimeout(20, TimeUnit.SECONDS)
-                .hostnameVerifier(noopHostnameVerifier.isPresent() && noopHostnameVerifier.get()
-                        ? (hostname, sslSession) -> true
-                        : OkHostnameVerifier.INSTANCE
-                )
-                .build();
+        String proxyHost = config.getString("ProxyHost").trim();
+        String proxyPort = config.getString("ProxyPort").trim();
+        final String authUser = config.getString("ProxyUser").trim();
+        final String authPassword = config.getString("ProxyPw").trim();
+
+        OkHttpClient httpClient; // initialized below
+        if (proxyHost != null && !proxyHost.isEmpty() && authPassword == null && authPassword.isEmpty()) {
+            System.setProperty("jdk.http.auth.tunneling.disabledSchemes", "");
```

Comment:
This property shouldn't be changed, it can be changed to be blank if needed by the user

Reply:
This is modified if proxy is enabled, see: https://stackoverflow.com/questions/41806422/java-web-start-unable-to-tunnel-through-proxy-since-java-8-update-111